library.py

```diff
@@ -43,46 +37,69 @@ def add_book(filename, isbn, title, author):
     # to the books object.
 
     books[isbn] = {'title': title, 'author':author}
-
+    
     # Finally, write code that writes the new data to the library
     # Do we need to return anything?
     with open(filename) as f:
         json.dump({'students':students,'books':books}, f)
-
+        
+    
 def remove_book(filename, isbn):
     students, books = open_library(filename)
 
     # How can we *remove* an item from a dictionary?
     # Write code to delete the book keyed by isbn in the space below
 
+    if 'isbn' in books:
+        del books['isbn']
     # Now write code that saves the new version of the data to your library
-
+    with open(filename) as f:
+        json.dump ({'students' : students, 'books' : books} ,f)
+        
 def check_out(filename, isbn, s_id):
     students, books = open_library(filename)
 
     # Find a way to mark a book as checked out. Be sure to associate
     # the book with the student who borrowed it!
-
+    
+    checked_out = {}
+    if 'isbn' in books:
+        checked_out.updated(books[isbn])
 
     # And again save the data here
-
+    with open(filename) as f:
+        json.dump({'students' : students, 'books' : books} ,f)
+        
     pass
 
 
 def return_book(filename, isbn):
     students, books = open_library(filename)
 
+    if 'isbn' in checked_out:
+        books.updated(checked_out)
+        
     # Now ensure that the book is no longer checked out and save the changes
     # to the library.
-
+    
+    with open(filename) as f:
+        json.dump({'students' : students, 'books' : books} ,f)
+    
     pass
 
 
 def status(filename):
     students, books = open_library(filename)
+    
+    if books[isbn] == True:
+        return book[isbn]
+    else:
+        return checked_out[isbn]
```

Comment:
`checked_out` isn't defined within the scope of this function. Are you sure you tested your code?